### PR TITLE
Sindhu/update_ngraph_api_call

### PR DESCRIPTION
**Pull request by sindhu-nervana**

### CODE REVIEW

**Comment by sindhu-nervana** [comment]
TESTNOW

File: src/ngraph_encapsulate_op.cc [hunk 1]
```diff
@@ -319,7 +319,7 @@ class NGraphEncapsulateOp : public OpKernel {
         m_ng_functions.erase(m_lru.back());
 
         // Call delete function here pf he erased func
-        auto exec = op_backend->compile(evicted_ng_function);
+        auto exec = m_exec_map.find(evicted_ng_function)->second;
         op_backend->remove_compiled_function(exec);
```
Comment (shresthamalik) [inline src/ngraph_encapsulate_op.cc]: Need to delete exec from the m_exec_map_

File: src/ngraph_encapsulate_op.cc [hunk 2]
```diff
@@ -350,15 +350,37 @@ class NGraphEncapsulateOp : public OpKernel {
                        << " Output Tensors freed: "
                        << output_tensors_bytes_free / (1024 * 1024) << " MB";
       }
-      m_ng_functions[signature] = ng_function;
+
+      try {
+        ng_exec = op_backend->compile(ng_function);
+      } catch (const std::exception& exp) {
+        BackendManager::UnlockBackend(m_op_backend_name);
```
Comment (shresthamalik) [inline src/ngraph_encapsulate_op.cc]: We are unlocking without locking. Not sure if we need to lock before compiling.

File: src/ngraph_encapsulate_op.cc [hunk 3]
```diff
@@ -371,7 +393,7 @@ class NGraphEncapsulateOp : public OpKernel {
         m_lru.remove(signature);
         m_lru.push_front(signature);
       }
-      ng_function = it->second;
+      ng_exec = it->second;
```
Comment (shresthamalik) [inline src/ngraph_encapsulate_op.cc]: In this case, ng_function is not updated here, but used later
https://github.com/NervanaSystems/ngraph-tf/pull/443/files#diff-4d55399542be3550bdfca644b849afa2R486

File: src/ngraph_encapsulate_op.cc (hunk 2 above)
Comment (sindhu-nervana) [reply src/ngraph_encapsulate_op.cc]: Added locking before unlocking.

**Comment by sindhu-nervana** [comment]
TESTNOW

File: CMakeLists.txt [hunk 4]
```diff
@@ -239,7 +239,7 @@ if (NOT USE_PRE_BUILT_NGRAPH)
     ExternalProject_Add(
         ext_ngraph
         GIT_REPOSITORY https://github.com/NervanaSystems/ngraph
-        GIT_TAG d3453447883fe48e63126316d5da72fcbd47dded
+        GIT_TAG df9e23761f5f7a7a7d0134c8ae0c92a8bad2e961
```
Comment (avijit-nervana) [inline CMakeLists.txt]: This should be set to release tag i.e., `v0.15.0-rc.1` at this point. I will add this change.

File: bazel/WORKSPACE [hunk 5]
```diff
@@ -24,11 +24,11 @@ tf_configure(
 new_http_archive(
     name = "ngraph",
     build_file = "bazel/ngraph.BUILD",
-    sha256 = "741812c14c70882e1ddb28e27c4a5e890b5fef3b2ec81ab6662dcd47127e9a35",
-    strip_prefix = "ngraph-d3453447883fe48e63126316d5da72fcbd47dded",
+    sha256 = "ec8c8933d7e6f654cf5b316475f67b6934c019f00144e9999f9b4877a53b2a1f",
+    strip_prefix = "ngraph-df9e23761f5f7a7a7d0134c8ae0c92a8bad2e961",
```
Comment (avijit-nervana) [inline bazel/WORKSPACE]: This should be set to release tag i.e., `v0.15.0-rc.1` at this point. I will add this change.

File: build_ngtf.py [hunk 6]
```diff
@@ -432,7 +432,7 @@ def main():
     #-------------------------------
 
     # Component versions
-    ngraph_version = "d3453447883fe48e63126316d5da72fcbd47dded"
+    ngraph_version = "df9e23761f5f7a7a7d0134c8ae0c92a8bad2e961"
```
Comment (avijit-nervana) [inline build_ngtf.py]: This should be set to release tag i.e., `v0.15.0-rc.1` at this point. I will add this change.

**Pull request merged**